MapboxGeocoder/MBGeocoder.swift

```diff
@@ -269,8 +280,25 @@ open class Geocoder: NSObject {
                     completionHandler(data)
                 }
             } catch {
-                DispatchQueue.main.async {
-                    errorHandler(error as NSError)
+                // Handle single & single batch geocoding queries
+                do {
+                    let result = try decoder.decode(GeocodeAPIResult.self, from: data)
+                    // Check if geocoding query failed
+                    guard result.message == nil else {
```

Comment:
Nit: We can also turn this `guard` statement into an `if let` so that the line below doesn’t have to force-unwrap `result.message`.